crates/solver/src/in_flight_orders.rs

```diff
@@ -70,11 +70,16 @@ impl InFlightOrders {
                 // fill-or-kill orders can only be used once and there is already a trade in flight
                 // for this one => Modify it such that it gets filtered out in the next step.
                 order.metadata.executed_buy_amount = u256_to_big_uint(&order.data.buy_amount);
+                order.metadata.executed_sell_amount = u256_to_big_uint(&order.data.sell_amount);
             }
         });
-        auction.orders.retain(|order| {
-            u256_to_big_uint(&order.data.buy_amount) > order.metadata.executed_buy_amount
-                && u256_to_big_uint(&order.data.sell_amount) > order.metadata.executed_sell_amount
+        auction.orders.retain(|order| match order.data.kind {
+            OrderKind::Sell => {
+                u256_to_big_uint(&order.data.sell_amount) > order.metadata.executed_sell_amount
```

Comment:
I think it should be this:

```suggestion
                u256_to_big_uint(&order.data.sell_amount) > order.metadata.executed_sell_amount_before_fees
```